### PR TITLE
✨[amp-analytics] Session Count for Session Analytics

### PR DESCRIPTION
**Pull request by micajuine-ho**
Partial for #33990

### CODE REVIEW

File: extensions/amp-analytics/0.1/session-manager.js [hunk 1]
```diff
@@ -166,7 +175,10 @@ export class SessionManager {
  * @return {boolean}
  */
 function isSessionExpired(session) {
-  return session.lastAccessTimestamp + SESSION_MAX_AGE_MILLIS < Date.now();
+  return (
+    session[SESSION_VALUES.ACCESS_TIMESTAMP] + SESSION_MAX_AGE_MILLIS <
+    Date.now()
+  );
```
Comment (alanorozco) [inline extensions/amp-analytics/0.1/session-manager.js:181]: nit:
```suggestion
  const accessTimestamp = session[SESSION_VALUES.ACCESS_TIMESTAMP];
  return accessTimestamp + SESSION_MAX_AGE_MILLIS < Date.now();
```

**Review by alanorozco — APPROVED** [review]

File: extensions/amp-analytics/0.1/session-manager.js [hunk 2]
```diff
@@ -120,31 +113,51 @@ export class SessionManager {
...
+  /**
+   * Check if session has expired and reset values (id, count) if so.
+   * Also update `lastAccessTimestamp`.
+   * @param {!SessionInfoDef} session
+   * @return {!SessionInfoDef}
+   */
+  updateSession_(session) {
+    if (isSessionExpired(session)) {
+      const newSessionCount =
+        session.count === undefined ? 1 : session.count + 1;
```
[the remark this answers is not in the document]
Comment (micajuine-ho) [reply extensions/amp-analytics/0.1/session-manager.js]: Only in the rare case of a user using session analytics before this PR comes out.

File: extensions/amp-analytics/0.1/session-manager.js [hunk 3]
```diff
@@ -187,37 +200,41 @@ function getStorageKey(type) {
...
+ * @param {number} count
+ * @param {number|undefined} opt_lastAccessTimestamp
  * @return {!SessionInfoDef}
  */
 function constructSessionInfo(
   sessionId,
   creationTimestamp,
-  lastAccessTimestamp
+  count,
+  opt_lastAccessTimestamp
```
[the remark this answers is not in the document]
Comment (micajuine-ho) [reply extensions/amp-analytics/0.1/session-manager.js]: Discussed offline (did some refactoring for this PR and yes it should be).

**Comment by micajuine-ho** [comment]
cc @ampproject/wg-performance Can I get a approval for bundle size check?

File: extensions/amp-analytics/0.1/session-manager.js [hunk 4]
```diff
@@ -106,48 +98,64 @@ export class SessionManager {
...
+  /**
+   * Check if session has expired and reset values (id, count) if so.
+   * Also update `accessTimestamp`.
+   * @param {!SessionInfoDef} session
+   * @return {!SessionInfoDef}
+   */
+  updateSession_(session) {
+    const currentCount = session[SESSION_VALUES.COUNT];
+    if (isSessionExpired(session)) {
+      const newSessionCount = currentCount === undefined ? 1 : currentCount + 1;
```
Comment (rcebulko) [inline extensions/amp-analytics/0.1/session-manager.js:147]: readability suggestion:
```suggestion
      const newSessionCount = (currentCount ?? 0) + 1;
```

File: extensions/amp-analytics/0.1/session-manager.js [hunk 5]
```diff
@@ -186,38 +195,36 @@ function getStorageKey(type) {
 }
 
 /**
- * @param {SessionInfoDef|string} storedSession
- * @return {SessionInfoDef|undefined}
+ * @param {*} storedSession
```
Comment (rcebulko) [inline extensions/amp-analytics/0.1/session-manager.js:198]: When might this be of a different type?

**Review by rcebulko — APPROVED** [review]
left a couple comments but approving for bundle-size

File: extensions/amp-analytics/0.1/session-manager.js (hunk 5 above)
Comment (micajuine-ho) [reply extensions/amp-analytics/0.1/session-manager.js:198]: That's the type we get back from `storage-impl/get()`. I imagine it has to do with the fact that we can't guarantee that no one will tamper with LS.

**Pull request merged**